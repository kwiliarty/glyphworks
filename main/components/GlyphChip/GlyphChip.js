import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Glyph = styled.div`
  width: ${ props => props.width };
  min-width: ${ props => props.width };
  height: ${ props => props.width };
  border-radius: ${ props => props.theme.borderRadius };
  background-color: ${ props => props.theme.colors[props.bColor] };
  color: ${ props => props.theme.colors[props.color] };
  margin: 0.5rem;
  padding-top: 0.1em;
  padding-bottom: 0.2em;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: bold;
  font-size: calc(${ props => props.width } * 0.8);
`

const GlyphChip = props => {
  const { glyph } = props
  return (
    <Glyph { ...props }>
      { glyph }
    </Glyph>
  )
}

GlyphChip.propTypes = {
  /** A background color name */
  bColor: PropTypes.string,
  /** A foreground color name */
  color: PropTypes.string,
  /** A string to be displayed as a glyph */
  glyph: PropTypes.string,
  /** A CSS size description */
  width: PropTypes.string,
}

GlyphChip.defaultProps = {
  bColor: 'darkRed',
  color: 'veryLightBrown',
  glyph: '',
  width: '5rem',
}

export default GlyphChip
