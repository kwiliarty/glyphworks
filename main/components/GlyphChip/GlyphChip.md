A `GlyphChip` displays a glyph against a colored background.

```jsx
<div style={{ display: 'flex' }}>
  <GlyphChip glyph='p' />
  <GlyphChip glyph='t' width='2rem' />
  <GlyphChip glyph='k' color='veryLightGreen' bColor='darkGreen' />
</div>
```
