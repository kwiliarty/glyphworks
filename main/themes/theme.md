## Settings

```jsx noeditor
import theme from './glyphworks'

;
<dl>
  {
    Object.keys( theme ).map( key => {
      if ( typeof( theme[ key ] ) == 'string' ) {
        return (
          <React.Fragment key={ key } >
            <dt><strong>{ key }</strong></dt>
            <dd><pre>{ theme[ key ] }</pre></dd>
          </React.Fragment>
        )
      }
    })
  }
</dl>
```
