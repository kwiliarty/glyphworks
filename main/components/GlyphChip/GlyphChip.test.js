import React from 'react'
import GlyphChip from './GlyphChip'
import { render, screen } from 'utils'
import theme from 'glyphworks'

describe( 'A GlyphChip', () => {
  it( 'features a glyph', () => {
    render(
      <GlyphChip glyph='p' />
    )
    const chip = screen.getByText( 'p' )
    expect( chip ).toBeInTheDocument()
    expect( chip ).toMatchSnapshot()
  })
  it( 'accepts a width prop', () => {
    render(
      <GlyphChip glyph='t' width='3rem' />
    )
    const chip = screen.getByText( 't' )
    expect( chip ).toHaveStyleRule( 'width', '3rem' )
    expect( chip ).toHaveStyleRule( 'height', '3rem' )
    expect( chip ).toHaveStyleRule( 'height', '3rem' )
    expect( chip ).toMatchSnapshot()
  })
  it( 'accepts color and bColor props', () => {
    render(
      <GlyphChip glyph='k' color='darkGreen' bColor='veryLightGreen' />
    )
    const chip = screen.getByText( 'k' )
    expect( chip ).toHaveStyleRule( 'color', theme.colors.darkGreen )
    expect( chip ).toHaveStyleRule( 'background-color', theme.colors.veryLightGreen )
    expect( chip ).toMatchSnapshot()
  })
})
