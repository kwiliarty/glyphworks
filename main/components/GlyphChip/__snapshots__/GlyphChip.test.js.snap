// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A GlyphChip accepts a width prop 1`] = `
.c0 {
  width: 3rem;
  min-width: 3rem;
  height: 3rem;
  border-radius: 0.1em;
  background-color: #621818;
  color: #fff7d4;
  margin: 0.5rem;
  padding-top: 0.1em;
  padding-bottom: 0.2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  font-size: calc(3rem * 0.8);
}

<div
  class="c0"
  color="veryLightBrown"
  width="3rem"
>
  t
</div>
`;

exports[`A GlyphChip accepts color and bColor props 1`] = `
.c0 {
  width: 5rem;
  min-width: 5rem;
  height: 5rem;
  border-radius: 0.1em;
  background-color: #badfba;
  color: #134e13;
  margin: 0.5rem;
  padding-top: 0.1em;
  padding-bottom: 0.2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  font-size: calc(5rem * 0.8);
}

<div
  class="c0"
  color="darkGreen"
  width="5rem"
>
  k
</div>
`;

exports[`A GlyphChip features a glyph 1`] = `
.c0 {
  width: 5rem;
  min-width: 5rem;
  height: 5rem;
  border-radius: 0.1em;
  background-color: #621818;
  color: #fff7d4;
  margin: 0.5rem;
  padding-top: 0.1em;
  padding-bottom: 0.2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  font-size: calc(5rem * 0.8);
}

<div
  class="c0"
  color="veryLightBrown"
  width="5rem"
>
  p
</div>
`;
